### PR TITLE
create correct VCard and return correct error codes

### PR DESCRIPTION
**Pull request by schiessle**
- fix vcard creation to avoid having multiple "uids"
- fix controller middleware to only process related exceptions

Both should be backported to stable11

### CODE REVIEW

**Comment by nickvergessen** [comment]
Looks good, but are there any docs for the last commit? the change looks odd?!

File: apps/dav/lib/CardDAV/Converter.php [hunk 1]
```diff
@@ -71,7 +78,7 @@ public function createCardFromUser(IUser $user) {
 						break;
 					case AccountManager::PROPERTY_AVATAR:
 						if ($image !== null) {
-							$vCard->add('PHOTO', 'data:'.$image->mimeType().';base64,' . base64_encode($image->data()));
+							$vCard->add('PHOTO', $image->data(), ['ENCODING' => 'b', 'TYPE' => $image->mimeType()]);
```
Comment (rullzer) [inline apps/dav/lib/CardDAV/Converter.php]: No when vcards hold binary data bad things happen

**Review by rullzer — CHANGES_REQUESTED** [review]

File: apps/dav/lib/CardDAV/Converter.php (hunk 1 above)
Comment (schiessle) [inline apps/dav/lib/CardDAV/Converter.php]: sabreDAV encodes it as base64 automatically, check the result in the db after running: `sudo -u www-data ./occ dav:sync-system-addressbook`

Also see the corresponding tests which check for base64 encoded representation and not for binary data:  https://github.com/nextcloud/server/pull/3825/files#diff-4a7b91a5b6e2e8b8735b9e8c5603fcdbR149

Comment (rullzer) [inline apps/dav/lib/CardDAV/Converter.php]: Aaah my bad. I was under the impression we were doing vcard v4. :+1:

**Review by rullzer — APPROVED** [review]

**Comment by nickvergessen** [comment]
Tested and works with self signed certs now after my last fix.

**Review by nickvergessen — APPROVED** [review]

File: .htaccess [hunk 2]
```diff
@@ -79,3 +79,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 /core/templates/403.php
+ErrorDocument 404 /core/templates/404.php
```
Comment (LukasReschke) [inline .htaccess]: create

**Review by LukasReschke — CHANGES_REQUESTED** [review]
Also: "Your test case is not allowed to access the database." :)

File: apps/dav/lib/CardDAV/SyncService.php [hunk 3]
```diff
@@ -142,6 +137,24 @@ public function ensureSystemAddressBookExists($principal, $id, $properties) {
 	}
 
 	/**
+	 * Check if there is a valid certPath we should use
+	 */
+	protected function getCertPath() {
```
Comment (nickvergessen) [inline apps/dav/lib/CardDAV/SyncService.php]: Since this is get, return it with this method?

**Comment by schiessle** [comment]
stable11: https://github.com/nextcloud/server/pull/4029

**Pull request merged**